### ISSUE TITLE
False positive `length-zero-no-unit` on `fr` unit

### ISSUE THREAD

**Issue by dw-buildit** [stylelint/stylelint#4250]
> Clearly describe the bug

There is a false positive for `0fr` when used in multiple values for `grid-template-rows` or similar css-grid properties.

`0fr 1fr` behaves differently to `0 1fr` on this property. The `fr` unit refers to _remaining space_ in the grid. `0` (without a unit) is treated as no height at all.
Using `0 1fr` the second row will overlap the first. Using `0fr 1fr` the height of the first row will be taken in to account and the second row will not overlap. [Here's](https://codepen.io/djfw/pen/ExYbExz) a minimal codepen to display the difference between the two and why this is a false positive.

> Which rule, if any, is the bug related to?

`length-zero-no-unit`

> What CSS is needed to reproduce the bug?

```css
.grid {
  display: grid;
  grid-template-rows: 0fr 1fr;
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "length-zero-no-unit": true
  }
}
```

> Which version of stylelint are you using?

`10.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

`stylelint 'src/sass/**/*.scss'`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

I am using scss and `stylelint-scss@3.9.4` but I'm not sure if that matters here (could be wrong of course 🤷‍♂)

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

`Unexpected unit (length-zero-no-unit)`

**Comment by evilebottnawi** [comment]
We should ignore some properties like `grid-template-rows`/`line-height`/etc
